### ISSUE TITLE
Whirpool GUI over TOR

### ISSUE THREAD

**Issue by atom686** [mynodebtc/mynode#449]
Currently it's impossible to use Whirpool GUI if you are not on a local network.

It would be nice to be able to connect to Whirpool GUI over the TOR.

**Comment by tehelsper** [comment]
It actually should be possible. Port 8899 is available via tor using the onion url most services use and can be found on the Tor page. I'll add it to the list in the next update.

**Comment by atom686** [comment]
Nevertheless, it doesn't work. "Connection failed: Could not connect to CLI (may take a few seconds to start...)"

**Comment by tehelsper** [comment]
Are you using Windows? I had heard there was a tor issue with whirlpool on Windows at some point due to a firewall.

It works fine on my Linux machine.

Also, try using tor browser to visit https://<onion url>:8899 - the service is running fine if it responds with an API error message.

**Comment by atom686** [comment]
I'm using Mac.

Here's what I get when visiting through browser:
"Whitelabel Error Page

This application has no explicit mapping for /error, so you are seeing this as a fallback.
Sun Feb 28 20:56:01 CST 2021
There was an unexpected error (type=Forbidden, status=403).
Access Denied"

**Comment by xxi000000** [comment]
I'm having the same issue on macOS.

Do you need to have a Tor proxy running locally on your machine? I have Tor Browser, but neither `socks5://127.0.0.1:9150` nor `socks5://127.0.0.1:9050` works. Not sure if that's the issue, but that would be my guess.

**Comment by tehelsper** [comment]
Yeah, you need tor proxy running. I think tor browser may run a proxy for other apps, but I'm not sure.

Can you try this?

https://kremalicious.com/simple-tor-setup-on-mac-os-x#install-tor

**Comment by xxi000000** [comment]
Hmmm, gave that a try and it still doesn't work.

Just to confirm, I'm putting in the `https://[...].onion:8899`  hidden service URL as the CLI address, `socks5://127.0.0.1:9050` as the tor proxy, and my Whirlpool API key.

**Comment by Jacksper13** [comment]
I'm struggling with this too... I tried using socks5://127.0.0.1:9050 while running tor, and socks5://127.0.0.1:9150 while running Tor Browser. Both work, because the error passes from "Could not connect to CLI" to "Connection failed: Unexpected token B in JSON at position 0". Running ubuntu 20.04.

Inputting an invalid tor address, or not running the proxies fails with "Connection failed: Could not connect to CLI (may take a few seconds to start...)", so there's something going on in the connection that fails.

Mynode v0.2.53.

Can you look into this? Thanks!

**Comment by goodluck456** [comment]
Follow the instructions from this video on how to install it on Ubuntu: https://www.youtube.com/watch?v=Fj94nIrlX60
You can also install whirlpool-gui and whirlpool-client-gui in Windows 10/11. But first, go to Control Panel > Programs > Turn Windows features on and off > Turn on Windows Subsystem for Linux, click Okay and restart the computer. When you reboot and log back in, install Ubuntu from the Microsoft Store. Open Ubuntu and create a username and password. Then change the terminal text editor from nano to vim.basic. Then modify the /etc/sudoers file so your username gets root privileges.  You have to add the username below admin user like below syntax. These links show you how to do it: 
https://stackoverflow.com/questions/47806576/linux-username-is-not-in-the-sudoers-file-this-incident-will-be-reported
https://www.digitalocean.com/community/tutorials/how-to-edit-the-sudoers-file
If you don't know how to use Vim, this video will show you how to use it: https://www.youtube.com/watch?v=ggSyF1SVFr4
Once you've done that, then do `sudo apt-get update` followed by `sudo apt-get upgrade`. You may also have to do `sudo apt --fix-broken install` to reinstall missing dependencies, especially for whirlpool-gui. Once you've done that, install tor by `sudo apt-get install tor`. Once you've done all that, just follow along with the video I linked. It'll show you how to install whirlpool-gui and whirlpool-client-cli in a Ubuntu environment.
